src/drivers/opfs.ts

```diff
@@ -0,0 +1,109 @@
+import { defineDriver } from "./utils";
+import {
+  DRIVER_NAME,
+  exists,
+  getFileObject,
+  joinPaths,
+  normalizePath,
+  readFile,
+  readdirRecursive,
+  remove,
+  removeChildren,
+  unlink,
+  writeFile,
+} from "./utils/opfs-utils";
+
+export interface OPFSStorageOptions {
+  /**
+   * The filesystem root to use
+   * Defaults to the OPFS root at `navigator.storage.getDirectory()`
+   */
+  fs?: FileSystemDirectoryHandle | Promise<FileSystemDirectoryHandle>;
+
+  /**
+   * The base path to use for all operations
+   * Defaults to the root directory (empty string)
+   */
+  base?: string;
+
+  /**
+   * A callback to ignore certain files in getKeys()
+   */
+  ignore?: (path: string) => boolean;
+
+  /**
+   * Whether to ignore any write operations
+   */
+  readOnly?: boolean;
+
+  /**
+   * Whether to disallow clearing the storage
+   */
+  noClear?: boolean;
```

Comment:
These two options are nice but i introduced them to main `fs` mainly because of security/legacy reasons of unintentional filesystem removals. If not really necessary, i would prefer to not introduce them in first step but later think of a generic unstorage option for readonly mountpoints supporting for all drivers.

Reply:
Removed `noClear` and `readOnly` options. A generic options sounds like a good idea for this.